src/nixdataarray.cc

```diff
@@ -52,8 +52,35 @@ namespace nixdataarray {
         nix::DataArray da = input.entity<nix::DataArray>(1);
         nix::DataType dtype = input.dtype(2);
 
-        nix::NDSize count = input.ndsize(2);
-        nix::NDSize offset(0);
+		const mxArray *arr = input.get_mx_array(2);
+
+		mwSize ndims = mxGetNumberOfDimensions(arr);
+		const mwSize *dims = mxGetDimensions(arr);
+        
+		mwSize tmp = ndims;
+		
+		// Quick fix for writing data arrays that have exactly one row.
+		// This fix does not resolve more complex arrays that have a 
+		// last dimension of exactly one row.
+		bool skip = false;
+		if (dims[ndims-1] == 1) {
+			skip = true;
+			tmp = ndims - 1;
+		}
+		nix::NDSize count(tmp);
+
+		for (mwSize i = 0; i < ndims; i++) {
+			if (i != ndims - 1 || (i == ndims - 1 && !skip)) {
+				if (skip) {
```

Comment:
I must be missing something but could you just write `for (mwSize i = 0; i < tmp; i++)`?
Or maybe  just `ndims--` in line 67 and have the rest as it were?